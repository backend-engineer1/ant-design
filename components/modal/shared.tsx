import CloseOutlined from '@ant-design/icons/CloseOutlined';
import React from 'react';
import Button from '../button';
import { convertLegacyProps } from '../button/button';
import { useLocale } from '../locale';
import type { ModalProps } from './interface';
import { getConfirmLocale } from './locale';

export function renderCloseIcon(prefixCls: string, closeIcon?: React.ReactNode) {
  return (
    <span className={`${prefixCls}-close-x`}>
      {closeIcon || <CloseOutlined className={`${prefixCls}-close-icon`} />}
    </span>
  );
}

interface FooterProps {
  onOk?: React.MouseEventHandler<HTMLButtonElement | HTMLAnchorElement>;
  onCancel?: React.MouseEventHandler<HTMLButtonElement | HTMLAnchorElement>;
}

export const Footer: React.FC<
  FooterProps &
    Pick<
      ModalProps,
      | 'footer'
      | 'okText'
      | 'okType'
      | 'cancelText'
      | 'confirmLoading'
      | 'okButtonProps'
      | 'cancelButtonProps'
    >
> = (props) => {
  const {
    okText,
    okType = 'primary',
    cancelText,
    confirmLoading,
    onOk,
    onCancel,
    okButtonProps,
    cancelButtonProps,
  } = props;

  const [locale] = useLocale('Modal', getConfirmLocale());

  return (
    <>
      <Button onClick={onCancel} {...cancelButtonProps}>
        {cancelText || locale?.cancelText}
      </Button>
      <Button
        {...convertLegacyProps(okType)}
        loading={confirmLoading}
        onClick={onOk}
        {...okButtonProps}
      >
        {okText || locale?.okText}
      </Button>
    </>
  );
};
