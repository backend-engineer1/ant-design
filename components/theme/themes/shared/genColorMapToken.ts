import type { ColorMapToken, SeedToken } from '../../interface';
import type {
  GenerateBgPalettes,
  GenerateErrorPalettes,
  GenerateInfoPalettes,
  GeneratePrimaryPalettes,
  GenerateSuccessPalettes,
  GenerateTextAlphaPalettes,
  GenerateWarningPalettes,
} from '../IPalettes';

interface PaletteGenerators {
  generatePrimaryPalettes: GeneratePrimaryPalettes;
  generateSuccessPalettes: GenerateSuccessPalettes;
  generateErrorPalettes: GenerateErrorPalettes;
  generateWarningPalettes: GenerateWarningPalettes;
  generateInfoPalettes: GenerateInfoPalettes;
  generateTextAlphaPalettes: GenerateTextAlphaPalettes;
  generateBgPalettes: GenerateBgPalettes;
}

export default function genColorMapToken(
  seed: SeedToken,
  {
    generatePrimaryPalettes,
    generateSuccessPalettes,
    generateErrorPalettes,
    generateWarningPalettes,
    generateInfoPalettes,
    generateTextAlphaPalettes,
    generateBgPalettes,
  }: PaletteGenerators,
): ColorMapToken {
  const {
    colorSuccess: colorSuccessBase,
    colorWarning: colorWarningBase,
    colorError: colorErrorBase,
    colorInfo: colorInfoBase,
    colorPrimary: colorPrimaryBase,
    colorBgBase,
    colorTextBase,
  } = seed;

  const primaryColors = generatePrimaryPalettes(colorPrimaryBase);
  const successColors = generateSuccessPalettes(colorSuccessBase);
  const warningColors = generateWarningPalettes(colorWarningBase);
  const errorColors = generateErrorPalettes(colorErrorBase);
  const infoColors = generateInfoPalettes(colorInfoBase);
  const bgColors = generateBgPalettes(colorBgBase, colorTextBase);
  const textColors = generateTextAlphaPalettes(colorTextBase);

  return {
    ...bgColors,
    ...textColors,

    colorPrimaryBg: primaryColors[1],
    colorPrimaryBgHover: primaryColors[2],
    colorPrimaryBorder: primaryColors[3],
    colorPrimaryBorderHover: primaryColors[4],
    colorPrimaryHover: primaryColors[5],
    colorPrimary: primaryColors[6],
    colorPrimaryActive: primaryColors[7],
    colorPrimaryText: primaryColors[8],
    colorPrimaryTextHover: primaryColors[9],

    colorSuccessBg: successColors[1],
    colorSuccessBorder: successColors[3],
    colorSuccess: successColors[6],

    colorErrorBg: errorColors[1],
    colorErrorBorder: errorColors[3],
    colorErrorHover: errorColors[5],
    colorError: errorColors[6],
    colorErrorActive: errorColors[7],

    colorWarningBg: warningColors[1],
    colorWarningBorder: warningColors[3],
    colorWarningHover: warningColors[5],
    colorWarning: warningColors[6],
    colorWarningActive: warningColors[7],

    colorInfoBg: infoColors[1],
    colorInfoBorder: infoColors[3],
    colorInfo: infoColors[6],
  };
}
