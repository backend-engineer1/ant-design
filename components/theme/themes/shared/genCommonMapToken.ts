import type { CommonMapToken, SeedToken } from '../../interface';
import genFontSizes from './genFontSizes';
import genRadius from './genRadius';

export default function genCommonMapToken(token: SeedToken): CommonMapToken {
  const { motionUnit, motionBase, fontSize, borderRadius, lineWidth } = token;

  const fontSizes = genFontSizes(fontSize);

  return {
    // motion
    motionDurationFast: `${(motionBase + motionUnit).toFixed(1)}s`,
    motionDurationMid: `${(motionBase + motionUnit * 2).toFixed(1)}s`,
    motionDurationSlow: `${(motionBase + motionUnit * 3).toFixed(1)}s`,

    // font
    fontSizes: fontSizes.map((fs) => fs.size),
    lineHeights: fontSizes.map((fs) => fs.lineHeight),

    // line
    lineWidthBold: lineWidth + 1,

    // radius
    ...genRadius(borderRadius),
  };
}
