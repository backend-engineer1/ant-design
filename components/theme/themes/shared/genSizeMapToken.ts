import type { SeedToken, SizeMapToken } from '../../interface';

export default function genSizeMapToken(token: SeedToken): SizeMapToken {
  const { sizeUnit, sizeBaseStep } = token;

  return {
    sizeXXL: sizeUnit * (sizeBaseStep + 8), // 48
    sizeXL: sizeUnit * (sizeBaseStep + 4), // 32
    sizeLG: sizeUnit * (sizeBaseStep + 2), // 24
    sizeMD: sizeUnit * (sizeBaseStep + 1), // 20
    sizeMS: sizeUnit * sizeBaseStep, // 16
    size: sizeUnit * sizeBaseStep, // 16
    sizeSM: sizeUnit * (sizeBaseStep - 1), // 12
    sizeXS: sizeUnit * (sizeBaseStep - 2), // 8
    sizeXXS: sizeUnit * (sizeBaseStep - 3), // 4
  };
}
