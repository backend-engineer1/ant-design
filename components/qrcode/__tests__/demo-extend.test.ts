import { extendTest } from '../../../tests/shared/demoTest';

extendTest('qrcode');
