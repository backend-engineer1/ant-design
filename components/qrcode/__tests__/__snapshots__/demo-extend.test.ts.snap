// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders components/qrcode/demo/Popover.tsx extend context correctly 1`] = `
Array [
  <img
    alt="icon"
    height="100"
    src="https://gw.alipayobjects.com/zos/rmsportal/KDpgvguMpGfqaHPjicRK.svg"
    width="100"
  />,
  <div
    class="ant-popover ant-zoom-big-appear ant-zoom-big-appear-prepare ant-zoom-big ant-popover-placement-top"
    style="left: -1000vw; top: -1000vh; box-sizing: border-box;"
  >
    <div
      class="ant-popover-arrow"
      style="position: absolute; bottom: 0px; left: 0px;"
    />
    <div
      class="ant-popover-content"
    >
      <div
        class="ant-popover-inner"
        role="tooltip"
        style="padding: 0px;"
      >
        <div
          class="ant-popover-inner-content"
        >
          <div
            class="ant-qrcode ant-qrcode-borderless"
            style="width: 160px; height: 160px;"
          >
            <canvas
              height="134"
              style="height: 134px; width: 134px;"
              width="134"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`renders components/qrcode/demo/base.tsx extend context correctly 1`] = `
<div
  class="ant-qrcode"
  style="width: 160px; height: 160px;"
>
  <canvas
    height="134"
    style="height: 134px; width: 134px;"
    width="134"
  />
</div>
`;

exports[`renders components/qrcode/demo/customColor.tsx extend context correctly 1`] = `
<div
  class="ant-space ant-space-horizontal ant-space-align-center"
>
  <div
    class="ant-space-item"
    style="margin-right: 8px;"
  >
    <div
      class="ant-qrcode"
      style="margin-bottom: 16px; background-color: rgb(245, 245, 245); width: 160px; height: 160px;"
    >
      <canvas
        height="134"
        style="height: 134px; width: 134px;"
        width="134"
      />
    </div>
  </div>
  <div
    class="ant-space-item"
  >
    <div
      class="ant-qrcode"
      style="margin-bottom: 16px; background-color: rgb(245, 245, 245); width: 160px; height: 160px;"
    >
      <canvas
        height="134"
        style="height: 134px; width: 134px;"
        width="134"
      />
    </div>
  </div>
</div>
`;

exports[`renders components/qrcode/demo/customSize.tsx extend context correctly 1`] = `
Array [
  <div
    class="ant-btn-group"
    style="margin-bottom: 16px;"
  >
    <button
      class="ant-btn ant-btn-default"
      type="button"
    >
      <span
        aria-label="minus"
        class="anticon anticon-minus"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="minus"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M872 474H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h720c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
          />
        </svg>
      </span>
      <span>
        Smaller
      </span>
    </button>
    <button
      class="ant-btn ant-btn-default"
      type="button"
    >
      <span
        aria-label="plus"
        class="anticon anticon-plus"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="plus"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <defs>
            <style />
          </defs>
          <path
            d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
          />
          <path
            d="M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z"
          />
        </svg>
      </span>
      <span>
        Larger
      </span>
    </button>
  </div>,
  <div
    class="ant-qrcode"
    style="width: 160px; height: 160px;"
  >
    <canvas
      height="134"
      style="height: 134px; width: 134px;"
      width="134"
    />
    <img
      src="https://gw.alipayobjects.com/zos/rmsportal/KDpgvguMpGfqaHPjicRK.svg"
      style="display: none;"
    />
  </div>,
]
`;

exports[`renders components/qrcode/demo/download.tsx extend context correctly 1`] = `
<div
  id="myqrcode"
>
  <div
    class="ant-qrcode"
    style="margin-bottom: 16px; width: 160px; height: 160px;"
  >
    <canvas
      height="134"
      style="height: 134px; width: 134px;"
      width="134"
    />
  </div>
  <button
    class="ant-btn ant-btn-primary"
    type="button"
  >
    <span>
      Download
    </span>
  </button>
</div>
`;

exports[`renders components/qrcode/demo/errorlevel.tsx extend context correctly 1`] = `
Array [
  <div
    class="ant-qrcode"
    style="margin-bottom: 16px; width: 160px; height: 160px;"
  >
    <canvas
      height="134"
      style="height: 134px; width: 134px;"
      width="134"
    />
  </div>,
  <div
    class="ant-segmented"
  >
    <div
      class="ant-segmented-group"
    >
      <label
        class="ant-segmented-item ant-segmented-item-selected"
      >
        <input
          checked=""
          class="ant-segmented-item-input"
          type="radio"
        />
        <div
          class="ant-segmented-item-label"
          title="L"
        >
          L
        </div>
      </label>
      <label
        class="ant-segmented-item"
      >
        <input
          class="ant-segmented-item-input"
          type="radio"
        />
        <div
          class="ant-segmented-item-label"
          title="M"
        >
          M
        </div>
      </label>
      <label
        class="ant-segmented-item"
      >
        <input
          class="ant-segmented-item-input"
          type="radio"
        />
        <div
          class="ant-segmented-item-label"
          title="Q"
        >
          Q
        </div>
      </label>
      <label
        class="ant-segmented-item"
      >
        <input
          class="ant-segmented-item-input"
          type="radio"
        />
        <div
          class="ant-segmented-item-label"
          title="H"
        >
          H
        </div>
      </label>
    </div>
  </div>,
]
`;

exports[`renders components/qrcode/demo/icon.tsx extend context correctly 1`] = `
<div
  class="ant-qrcode"
  style="width: 160px; height: 160px;"
>
  <canvas
    height="134"
    style="height: 134px; width: 134px;"
    width="134"
  />
  <img
    src="https://gw.alipayobjects.com/zos/rmsportal/KDpgvguMpGfqaHPjicRK.svg"
    style="display: none;"
  />
</div>
`;

exports[`renders components/qrcode/demo/status.tsx extend context correctly 1`] = `
<div
  class="ant-space ant-space-horizontal ant-space-align-center"
  style="flex-wrap: wrap; margin-bottom: -8px;"
>
  <div
    class="ant-space-item"
    style="margin-right: 8px; padding-bottom: 8px;"
  >
    <div
      class="ant-qrcode"
      style="width: 160px; height: 160px;"
    >
      <div
        class="ant-qrcode-mask"
      >
        <div
          aria-busy="true"
          aria-live="polite"
          class="ant-spin ant-spin-spinning"
        >
          <span
            class="ant-spin-dot ant-spin-dot-spin"
          >
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
            <i
              class="ant-spin-dot-item"
            />
          </span>
        </div>
      </div>
      <canvas
        height="134"
        style="height: 134px; width: 134px;"
        width="134"
      />
    </div>
  </div>
  <div
    class="ant-space-item"
    style="padding-bottom: 8px;"
  >
    <div
      class="ant-qrcode"
      style="width: 160px; height: 160px;"
    >
      <div
        class="ant-qrcode-mask"
      >
        <p
          class="ant-qrcode-expired"
        >
          QR code expired
        </p>
        <button
          class="ant-btn ant-btn-link"
          type="button"
        >
          <span
            aria-label="reload"
            class="anticon anticon-reload"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="reload"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M909.1 209.3l-56.4 44.1C775.8 155.1 656.2 92 521.9 92 290 92 102.3 279.5 102 511.5 101.7 743.7 289.8 932 521.9 932c181.3 0 335.8-115 394.6-276.1 1.5-4.2-.7-8.9-4.9-10.3l-56.7-19.5a8 8 0 00-10.1 4.8c-1.8 5-3.8 10-5.9 14.9-17.3 41-42.1 77.8-73.7 109.4A344.77 344.77 0 01655.9 829c-42.3 17.9-87.4 27-133.8 27-46.5 0-91.5-9.1-133.8-27A341.5 341.5 0 01279 755.2a342.16 342.16 0 01-73.7-109.4c-17.9-42.4-27-87.4-27-133.9s9.1-91.5 27-133.9c17.3-41 42.1-77.8 73.7-109.4 31.6-31.6 68.4-56.4 109.3-73.8 42.3-17.9 87.4-27 133.8-27 46.5 0 91.5 9.1 133.8 27a341.5 341.5 0 01109.3 73.8c9.9 9.9 19.2 20.4 27.8 31.4l-60.2 47a8 8 0 003 14.1l175.6 43c5 1.2 9.9-2.6 9.9-7.7l.8-180.9c-.1-6.6-7.8-10.3-13-6.2z"
              />
            </svg>
          </span>
          <span>
            Refresh
          </span>
        </button>
      </div>
      <canvas
        height="134"
        style="height: 134px; width: 134px;"
        width="134"
      />
    </div>
  </div>
</div>
`;
